### PR TITLE
fix: resolve chained dependency handling and add Compose depends_on support

### PR DESCRIPTION
**Pull request by nicholas-fedor**
This PR fixes critical issues with container dependency handling, ensuring correct stop/start ordering for chained dependencies and adds automatic detection of Docker Compose depends_on relationships. The changes prevent service disruptions during updates by properly sequencing container operations according to dependency graphs.

## Problem

Watchtower failed to handle multi-level dependencies correctly, causing dependent containers to restart after their dependencies instead of before, leading to broken application states during updates. Additionally, Docker Compose's `depends_on` labels were not automatically recognized, requiring manual configuration. These issues, detailed in issue #1024, could result in data loss or service unavailability when updating interdependent containers.

## Solution

Implemented proper topological sorting using Kahn's algorithm with robust cycle detection to prevent infinite loops. Added automatic parsing and recognition of Docker Compose `depends_on` labels alongside existing Watchtower dependency labels. Refactored the sorting package for better maintainability and added comprehensive test coverage including edge cases and performance benchmarks.

## Changes

- **Fixed dependency ordering**: Corrected stop/start sequence to stop dependents before dependencies and restart dependencies before dependents
- **Added Compose depends_on support**: Automatic detection and parsing of Docker Compose `depends_on` relationships
- **Improved sorting algorithm**: Replaced fragile sorting with Kahn's algorithm and cycle detection
- **Enhanced documentation**: Comprehensive rewrite of linked containers documentation with examples and troubleshooting
- **Code refactoring**: Split sorting logic into dedicated files, improved error handling, and added extensive tests
- **Performance optimizations**: Better time sorting with pre-parsed timestamps and fallback handling

## Summary by CodeRabbit

* **New Features**
  * Multi-source dependency detection, service-name resolution, normalized container names, dependency-aware update/restart ordering, and a new "restarted" count exposed in API, metrics and notifications.

* **Bug Fixes**
  * Circular-dependency detection with warnings and skips; improved implicit linked-restart propagation and deterministic stop/start ordering.

* **Documentation**
  * Expanded linked-containers guide, rolling-restart advisory, API/metrics and notification template updates to include Restarted.

* **Tests**
  * Large addition of unit, benchmark and fuzz tests for dependency graphs, sorting, cycles and restart flows.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -273,7 +274,12 @@ func preRun(cmd *cobra.Command, _ []string) {
 
 	// Set additional configuration flags that control update behavior and scope.
 	enableLabel, _ = flagsSet.GetBool("label-enable")
+
 	disableContainers, _ = flagsSet.GetStringSlice("disable-containers")
+	for i := range disableContainers {
+		disableContainers[i] = util.NormalizeContainerName(disableContainers[i])
+	}
```
[the remark this answers is not in the document]
Comment (nicholas-fedor) [reply cmd/root.go:281]: `c.Name()` returns a normalized name.
